src/System.Security.Permissions/src/System/Security/Permissions/PrincipalPermission.cs

```diff
@@ -2,25 +2,471 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
+using System.Collections;
+using System.Collections.Generic;
+using System.Security.Principal;
+using System.Threading;
+
 namespace System.Security.Permissions
 {
-    //TODO #9641: Implement PrincipalPermission with real behavior
+    [Serializable]
+    internal partial class IDRole
```

Comment:
This is a holdover from a previous change that I rebased out. It doesn't need to be partial anymore.